pkg/catalog/types.go

```diff
@@ -0,0 +1,18 @@
+package catalog
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CatalogImage struct {
+	Name          string   `json:"name"`
```

Comment:
- Missing `TypeMeta` and `ObjectMeta`.
- `AllTags` and `NonHiddentTags` should be in a separate `CatalogImageSpec` struct  and in `Spec` field.
- no need for `Namespace` as it is already is in `ObjectMeta`